dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/VisitorOpsMemory.java

```diff
@@ -64,6 +93,9 @@ public Event visitOpLoad(SpirvParser.OpLoadContext ctx) {
         List<Event> events = new ArrayList<>();
         if (builder.getType(resultType) instanceof ArrayType arrayType) {
             Type elType = arrayType.getElementType();
+            if(!(elType instanceof BooleanType || elType instanceof IntegerType || elType instanceof FloatType)) {
```

Comment:
This kind of type check happens in at least 3 places. I think you should refactor this into a `isScalar` function